enhancements/network/ovn-kubernetes-ipsec.md

```diff
@@ -0,0 +1,298 @@
+- Will not provide any implementation for DPDK interfaces.
+- Will not provide an option to enable tunnels that have been configured with
+self-signed certificates or pre-shared keys.
+- Will not do any explicit performance optimization of the datapath. As the
+implementation of this enhancement will primarily require changes to the control
+and management path, it is expected that this will not alter the performance of
+the current OVS/Kernel IPsec implementation.
+- Will not provide any interface to allow a cluster operator to specify
+preferred encryption, integrity, authentication algorithms
+- Will not explicitly encrypt cluster management traffic. Cluster management
+traffic has already been encrypted using TLS.
+- Will not allow granular configuration of which nodes will have IPsec enabled
+(for example Namespace to Namespace encryption) or how IPsec is configured
+between specific nodes. This enhancement will enable IPsec between all worker
+nodes within the cluster indiscriminately.
+- Will not provide IPsec connectivity between any worker node that does not form
+part of the Kubernetes cluster. This enhancement is only for Node-to-Node
+(east-west) communication within the cluster.
+
+## Proposal
+
+The following changes are proposed to the [network.operator.openshift.io API](https://github.com/openshift/api/blob/master/operator/v1/types_network.go).
+
+```go
+// ovnKubernetesConfig contains the configuration parameters for networks
+// using the ovn-kubernetes network project
+type OVNKubernetesConfig struct {
+	// mtu is the MTU to use for the tunnel interface. This must be 100
+	// bytes smaller than the uplink mtu.
+	// Default is 1400
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	MTU *uint32 `json:"mtu,omitempty"`
+	// geneve port is the UDP port to be used by geneve encapulation.
+	// Default is 6081
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	GenevePort *uint32 `json:"genevePort,omitempty"`
+	// HybridOverlayConfig configures an additional overlay network for peers that are
+	// not using OVN.
+	// +optional
+  HybridOverlayConfig *HybridOverlayConfig `json:"hybridOverlayConfig,omitempty"`
+  // EnableIPsec enables IPsec encryption on all pod-pod traffic within the cluser. If unset, the plugin will default
+  // EnableIPsec to false.
+  // +optional
+  EnableIPsec *bool `json:"enableIPsec,omitempty"`
+}
+```
+
+This will result in an updated OVNKubernetes Operator configuration object:
+(.spec.defaultNetwork.ovnKubernetesConfig) which will enable cluster-wide
+configuration of IPsec at cluster installation time:
+
+spec:
+  defaultNetwork:
+    type: OVNKubernetes
+    ovnKubernetesConfig:
+      mtu: 1400
+      genevePort: 6081
+      **enableIPsec: true/false**
+
+### User Stories
+
+#### Story 1
+
+As a cluster operator, I want to be able to enable IPsec at cluster installation
+time in the OVNKubernetes plugin in order to encrypt, authenticate and ensure
+the integrity of all pod-pod traffic.
```

Comment:
This user story doesn't add anything to the proposal and can just be removed.

Reply:
Yes, seems a bit redundant.